example_envs/bookinfo/reviews.rs

```diff
@@ -37,26 +37,36 @@ impl SimElement for Reviews {
             }
             // forward requests/responses from productpage or reviews
             if rpc.headers.contains_key("src") {
-                let source = &rpc.headers["src"];
-                let dest: &str;
-                if source == "ratings-v1" {
-                    dest = "productpage-v1";
-                } else if source == "productpage-v1" {
-                    dest = "ratings-v1";
-                } else {
-                    panic!("Unexpected RPC source {:?}", source);
-                }
+                let dest = self.choose_destination(&rpc);
+                rpc.headers.insert("dest".to_string(), dest.clone());
                 rpc.headers
                     .insert("src".to_string(), self.core_node.id.to_string());
-                ret.push((rpc, dest.to_string()));
+                rpc.headers
+                    .insert("location".to_string(), "egress".to_string());
+                if self.core_node.plugin.is_some() {
```

Comment:
Same comments as above. This is all duplicated code.

Reply:
Fixed